zend_abstract_interface/config/config_stable_file.c

```diff
@@ -0,0 +1,137 @@
+        zai_config_id config_id;
+        zai_str config_name = {.ptr = ZSTR_VAL(key), .len = ZSTR_LEN(key)};
+
+        if (!zai_config_stable_file_find_entry(config_name, &config_id)) {
+            continue;
+        }
+
+        zai_config_memoized_entry *memoized = &zai_config_memoized_entries[config_id];
+        zend_ini_entry *entry = memoized->ini_entries[0];
+
+        zend_string *value_str = zend_string_init(value, strlen(value), false);
+        zend_alter_ini_entry_ex(entry->name, value_str, PHP_INI_USER, stage, 0);
+        zend_string_release(value_str);
+    } ZEND_HASH_FOREACH_END();
+}
+
+static void config_value_dtor(zval *zv) {
+    void *ptr = Z_PTR_P(zv);
+    pefree(ptr, 1);
+}
+
+void zai_config_stable_file_minit(void) {
+    ddog_Configurator *configurator = ddog_library_configurator_new(false, DDOG_CHARSLICE_C("php"));
+
+    char *file = getenv("_DD_TEST_LIBRARY_CONFIG_LOCAL_FILE");
+    if (file) {
+        ddog_CStr path = {.ptr = file, .length = strlen(file)+1}; // FIXME: +1 -> https://github.com/DataDog/libdatadog/pull/924
+        ddog_library_configurator_with_local_path(configurator, path);
+    }
+    file = getenv("_DD_TEST_LIBRARY_CONFIG_FLEET_FILE");
+    if (file) {
+        ddog_CStr path = {.ptr = file, .length = strlen(file)+1}; // FIXME: +1 -> https://github.com/DataDog/libdatadog/pull/924
+        ddog_library_configurator_with_fleet_path(configurator, path);
+    }
+
+    // FIXME: without the call to ddog_library_configurator_with_process_info,
+    // some AppSec's integration tests fails
+#define DDOG_SLICE_CHARSLICE(arr) \
+    ((ddog_Slice_CharSlice){.ptr = arr, .len = sizeof(arr) / sizeof(arr[0])})
+
+    ddog_CharSlice args[] = {
+        DDOG_CHARSLICE_C("/usr/bin/php"),
+    };
+
+    ddog_CharSlice envp[] = {
+        DDOG_CHARSLICE_C("FOO=BAR"),
+    };
+    ddog_ProcessInfo process_info = {
+        .args = DDOG_SLICE_CHARSLICE(args),
+        .envp = DDOG_SLICE_CHARSLICE(envp),
+        .language = DDOG_CHARSLICE_C("php")
+    };
+    ddog_library_configurator_with_process_info(configurator, process_info);
+    //
+
+    ddog_Result_VecLibraryConfig config_result = ddog_library_configurator_get(configurator);
+    if (config_result.tag == DDOG_RESULT_VEC_LIBRARY_CONFIG_OK_VEC_LIBRARY_CONFIG) {
+        config_values = pemalloc(sizeof(HashTable), 1);
+        zend_hash_init(config_values, 8, NULL, config_value_dtor, 1);
+
+        ddog_Vec_LibraryConfig configs = config_result.ok;
+        for (uintptr_t i = 0; i < configs.len; i++) {
+            const ddog_LibraryConfig *cfg = &configs.ptr[i];
+            ddog_CStr env_name = ddog_library_config_name_to_env(cfg->name);
+
+            char *value = pestrndup(cfg->value.ptr, cfg->value.length, 1);
+            if (!value) {
+                continue;
+            }
+            zend_hash_str_add_ptr(config_values, env_name.ptr, env_name.length + 1, value);  // FIXME: +1 -> https://github.com/DataDog/libdatadog/pull/924
```

Comment:
Looks like it's fixed now

Reply:
libdatadog updated

Reply:
@bwoebi the update breaks the generation of the bindings. It seems it comes from this change: https://github.com/DataDog/libdatadog/commit/93f1c5b02c072083dbe7eb177791016f6a7ce0ea#diff-c602f1bd972c7e4d5371d7171f0c9ee087410548c48d1d519e9ba52f3ff17b49R38-R40

Any idea how to fix this?

Reply:
Let me PR it to libdatadog.